components/amorphic/lib/session/establishContinuedServerSession.js

```diff
@@ -69,9 +69,15 @@ function establishContinuedServerSession(req, controllerPath, initObjectTemplate
 
     let sessionData = getSessionCache(path, req.session.id, true, sessions);
     let ourObjectTemplate = getObjectTemplate(controller);
+
     ourObjectTemplate.memSession = sessionData;
     ourObjectTemplate.reqSession = req.session;
     controller.__request = req;
+
+    // getter for the response object, adding it directly on the controller results in circular reference issues
+    controller.__getResponseObj = function __getResponseObj() {
```

Comment:
I'm curious  as to why this wouldn't result in circular reference issues as well then

Reply:
hey! so amorphic needs to be able to serialize/deserialize these controller objects, essentially just stringifying / JSON.parse-ing. if the response object lives directly on the controller, we run into issues trying to stringify because there are circular references within the response object. if however, it's simply a getter here and we just rely on whatever the state of this `res` variable is at run time, we're not going to include it in the controller parsing and we won't run into this type of problem.

also, a secondary considering, it may be a bit safer since it's not a real property on the controller, there's less room to try and access this in a context in which it doesn't belong.